### PR TITLE
Commerce 12719

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/COMMERCE-12719

### CODE REVIEW

File: modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/factory/CustomCheckoutStepCETImplFactoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+			properties, sourceCodeURL, unicodeProperties);
+	}
+
+	@Override
+	public void validate(
+			UnicodeProperties newTypeSettingsUnicodeProperties,
+			UnicodeProperties oldTypeSettingsUnicodeProperties)
+		throws PortalException {
+
+		/*
```
Comment (luca-pellizzon) [inline modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/factory/CustomCheckoutStepCETImplFactoryImpl.java:58]: Why this comment block?

File: modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CustomCheckoutStep.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+		throws Exception {
+
+		CommerceContext commerceContext =
+			(CommerceContext)actionRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		CommerceOrder commerceOrder = commerceContext.getCommerceOrder();
+
+		_portalCatapult.launch(
+			commerceOrder.getCompanyId(), Http.Method.POST, "",
```
Comment (luca-pellizzon) [inline modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CustomCheckoutStep.java:89]: I think you actually need an oAuth2ApplicationExternalReferenceCode to communicate with external apps. I think you need to add it in your configuration.

File: modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CustomCheckoutStep.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		throws Exception {
+
+		CommerceContext commerceContext =
+			(CommerceContext)httpServletRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		CommerceOrder commerceOrder = commerceContext.getCommerceOrder();
+
+		_portalCatapult.launch(
+			commerceOrder.getCompanyId(), Http.Method.GET, "",
```
Comment (luca-pellizzon) [inline modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CustomCheckoutStep.java:107]: Same here.

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Pull request closed without merging**